##Using Open Images V7

In [ ]:
!pip install fiftyone

  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 23.8/23.8 MB 66.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 43.3/43.3 kB 2.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 140.0/140.0 kB 8.8 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 120.8/120.8 kB 7.4 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 44.8/44.8 kB 2.5 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.6/61.6 kB 3.4 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 112.5/112.5 kB 6.6 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 74.8/74.8 kB 4.0 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.0/62.0 MB 9.4 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 6.9/6.9 MB 65.5 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 9.9/9.9 MB 72.7 MB/s eta 0:00:00
  

restart session

In [ ]:
import fiftyone as fo
import fiftyone.zoo as foz
import fiftyone.types as fot
from fiftyone import ViewField as F

open images mit Axe, Baseball bat, Chainsaw, Drill, Glove, Screwdriver, Hammer, Weapon, Knife, Scissors

In [ ]:
labels = ["Knife"]


train_dataset_oi = foz.load_zoo_dataset(
        "open-images-v7",
        split="train",
        label_types=["detections"],
        classes=labels,
        max_samples = 5000,
        shuffle=True
)

val_dataset_oi = foz.load_zoo_dataset(
        "open-images-v7",
        split="validation",
        label_types=["detections"],
        classes=labels,
        max_samples = 500,
        shuffle=True
)

INFO:fiftyone.zoo.datasets:Downloading split 'train' to '/root/fiftyone/open-images-v7/train' if necessary


Only found 610 (<5000) samples matching your requirements


Found 445 images, downloading the remaining 165


INFO:fiftyone.utils.openimages:Found 445 images, downloading the remaining 165


 100% |███████████████████| 165/165 [14.5s elapsed, 0s remaining, 11.9 files/s]      


INFO:eta.core.utils: 100% |███████████████████| 165/165 [14.5s elapsed, 0s remaining, 11.9 files/s]      


Dataset info written to '/root/fiftyone/open-images-v7/info.json'


INFO:fiftyone.zoo.datasets:Dataset info written to '/root/fiftyone/open-images-v7/info.json'


Loading 'open-images-v7' split 'train'


INFO:fiftyone.zoo.datasets:Loading 'open-images-v7' split 'train'


 100% |█████████████████| 610/610 [3.1s elapsed, 0s remaining, 202.0 samples/s]      


INFO:eta.core.utils: 100% |█████████████████| 610/610 [3.1s elapsed, 0s remaining, 202.0 samples/s]      


Dataset 'open-images-v7-train-5000' created


INFO:fiftyone.zoo.datasets:Dataset 'open-images-v7-train-5000' created


INFO:fiftyone.zoo.datasets:Downloading split 'validation' to '/root/fiftyone/open-images-v7/validation' if necessary


INFO:fiftyone.utils.openimages:Downloading 'https://storage.googleapis.com/openimages/2018_04/validation/validation-images-with-rotation.csv' to '/root/fiftyone/open-images-v7/validation/metadata/image_ids.csv'


INFO:fiftyone.utils.openimages:Downloading 'https://storage.googleapis.com/openimages/v5/class-descriptions-boxable.csv' to '/root/fiftyone/open-images-v7/validation/metadata/classes.csv'


INFO:fiftyone.utils.openimages:Downloading 'https://storage.googleapis.com/openimages/2018_04/bbox_labels_600_hierarchy.json' to '/tmp/tmpc4f1e54e/metadata/hierarchy.json'


INFO:fiftyone.utils.openimages:Downloading 'https://storage.googleapis.com/openimages/v5/validation-annotations-bbox.csv' to '/root/fiftyone/open-images-v7/validation/labels/detections.csv'


Only found 59 (<500) samples matching your requirements


INFO:fiftyone.utils.openimages:Downloading 59 images


 100% |█████████████████████| 59/59 [6.4s elapsed, 0s remaining, 11.7 files/s]      


INFO:eta.core.utils: 100% |█████████████████████| 59/59 [6.4s elapsed, 0s remaining, 11.7 files/s]      


Dataset info written to '/root/fiftyone/open-images-v7/info.json'


INFO:fiftyone.zoo.datasets:Dataset info written to '/root/fiftyone/open-images-v7/info.json'


Loading 'open-images-v7' split 'validation'


INFO:fiftyone.zoo.datasets:Loading 'open-images-v7' split 'validation'


 100% |███████████████████| 59/59 [994.3ms elapsed, 0s remaining, 59.8 samples/s]      


INFO:eta.core.utils: 100% |███████████████████| 59/59 [994.3ms elapsed, 0s remaining, 59.8 samples/s]      


Dataset 'open-images-v7-validation-500' created


INFO:fiftyone.zoo.datasets:Dataset 'open-images-v7-validation-500' created


Filter only relevant labels from the dataset

In [ ]:
train_dataset = train_dataset_oi.filter_labels(
    "ground_truth",
    F("label").is_in(labels)
)

val_dataset = val_dataset_oi.filter_labels(
    "ground_truth",
    F("label").is_in(labels)
)

counts = train_dataset_oi.count_values("ground_truth.detections.label")

for c in labels:
    print(f"{c}: {counts.get(c, 0)}")

print("______________________________")

counts = val_dataset_oi.count_values("ground_truth.detections.label")

for c in labels:
    print(f"{c}: {counts.get(c, 0)}")

Knife: 850
______________________________
Knife: 80


In [ ]:
train_dataset_oi.export(
        export_dir=f"./dataset",
        dataset_type=fot.YOLOv5Dataset,
        label_field="ground_truth",
        classes=labels,
        split="train"
)
val_dataset_oi.export(
        export_dir=f"./dataset",
        dataset_type=fot.YOLOv5Dataset,
        label_field="ground_truth",
        classes=labels,
        split="val"
)

   2% |/----------------|  13/610 [111.7ms elapsed, 5.1s remaining, 116.3 samples/s] 

/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Apple' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Kitchen knife' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Cutting board' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Dagger' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Person' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Fashion accessory' not in provided classes
  warnings.warn(msg

   7% |█|---------------|  44/610 [428.9ms elapsed, 5.5s remaining, 102.6 samples/s] 

/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Toy' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Sculpture' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Sword' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Vegetable' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Fork' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Drink' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.

  11% |█----------------|  70/610 [637.1ms elapsed, 4.9s remaining, 109.9 samples/s] 

/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Man' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Clothing' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Cream' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Tin can' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Human face' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Fruit' not in provided classes
  warnings.warn(msg)
/usr/local/lib/pytho

  16% |██|--------------| 100/610 [844.3ms elapsed, 4.3s remaining, 118.4 samples/s] 

/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Baked goods' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Orange' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Picture frame' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Beer' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Bottle' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Flower' not in provided classes
  warnings.warn(msg)
/usr/local/l

  21% |███--------------| 127/610 [1.0s elapsed, 4.0s remaining, 121.8 samples/s]    

/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Cake' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Wheel' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Tire' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Fast food' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Egg (Food)' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Mixing bowl' not in provided classes
  warnings.warn(msg)
/usr/local/lib/

  28% |████/------------| 171/610 [1.4s elapsed, 3.4s remaining, 136.1 samples/s]    

/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Wine glass' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Woman' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Tomato' not in provided classes
  warnings.warn(msg)


  32% |█████\-----------| 198/610 [1.6s elapsed, 3.2s remaining, 138.3 samples/s]    

/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Rifle' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Wok' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Seafood' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Garden Asparagus' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Glove' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Chisel' not in provided classes
  warnings.warn(msg)


  37% |██████/----------| 224/610 [1.8s elapsed, 3.0s remaining, 134.0 samples/s]    

/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Mango' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Fish' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Snack' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Bread' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Vase' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Rose' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-

  45% |███████|---------| 272/610 [2.1s elapsed, 2.5s remaining, 141.7 samples/s]    

/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Wine' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Squid' not in provided classes
  warnings.warn(msg)


  49% |████████---------| 296/610 [2.3s elapsed, 2.4s remaining, 133.0 samples/s]    

/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Dairy Product' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Pomegranate' not in provided classes
  warnings.warn(msg)


  60% |██████████-------| 365/610 [2.7s elapsed, 1.7s remaining, 149.3 samples/s]    

/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Axe' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Coconut' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Cucumber' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Mobile phone' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Human hand' not in provided classes
  warnings.warn(msg)


  75% |████████████\----| 456/610 [3.2s elapsed, 945.7ms remaining, 171.8 samples/s] 

/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Bell pepper' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Vehicle' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Salad' not in provided classes
  warnings.warn(msg)


  83% |██████████████---| 508/610 [3.5s elapsed, 620.3ms remaining, 169.5 samples/s] 

/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Tree' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Skateboard' not in provided classes
  warnings.warn(msg)


  92% |███████████████/-| 562/610 [3.9s elapsed, 277.1ms remaining, 176.1 samples/s] 

/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Can opener' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Insect' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Grapefruit' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Coin' not in provided classes
  warnings.warn(msg)


 100% |█████████████████| 610/610 [4.1s elapsed, 0s remaining, 181.1 samples/s]      


/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Goggles' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Glasses' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Sunglasses' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Car' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Platter' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Broccoli' not in provided classes
  warnings.warn(msg)
/usr/local/lib/p

Directory './dataset' already exists; export will be merged with existing files


   0% ||------------------|  0/59 [4.4ms elapsed, ? remaining, ? samples/s] 

/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Tool' not in provided classes
  warnings.warn(msg)


  56% |██████████/--------| 33/59 [105.8ms elapsed, 83.3ms remaining, 312.0 samples/s] 

/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Kitchen utensil' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Tart' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Cantaloupe' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Human head' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Zucchini' not in provided classes
  warnings.warn(msg)
/usr/local/lib/python3.13/dist-packages/fiftyone/utils/yolo.py:1154: UserWarning: Ignoring object with label 'Radish' not in provided classes
  warnings.warn(msg)
/usr/

 100% |███████████████████| 59/59 [200.6ms elapsed, 0s remaining, 294.1 samples/s]     


INFO:eta.core.utils: 100% |███████████████████| 59/59 [200.6ms elapsed, 0s remaining, 294.1 samples/s]     


# Fire Datasets

In [ ]:
import kagglehub

# Download latest version
path_A = kagglehub.dataset_download("sinchanashivanand/indoor-fire-and-smoke-detection-with-yolov8")

print("Path to dataset files:", path_A)

Using Colab cache for faster access to the 'indoor-fire-and-smoke-detection-with-yolov8' dataset.
Path to dataset files: /kaggle/input/indoor-fire-and-smoke-detection-with-yolov8


In [ ]:
import kagglehub

# Download latest version
pathy = kagglehub.dataset_download("pengbo00/home-fire-dataset")

print("Path to dataset files:", pathy)

Using Colab cache for faster access to the 'home-fire-dataset' dataset.
Path to dataset files: /kaggle/input/home-fire-dataset


0 = Feuer, 1 = Smoke

mege both datasets in coherent format and only take 1000 for train and 200 for val

In [ ]:
import os
import shutil
import random
from pathlib import Path

dataset_A = Path("/kaggle/input/home-fire-dataset")
dataset_B = Path("/kaggle/input/indoor-fire-and-smoke-detection-with-yolov8/data")
output = Path("/content/fire_dataset")

TRAIN_SIZE = 1000
VAL_SIZE = 200
SEED = 42

random.seed(SEED)

if output.exists():
    shutil.rmtree(output)

# out dir
for split in ["train", "val"]:
    (output / "images" / split).mkdir(parents=True, exist_ok=True)
    (output / "labels" / split).mkdir(parents=True, exist_ok=True)

samples = []

image_extensions = [".jpg", ".jpeg", ".png", ".bmp", ".webp"]

def collect_dataset(images_dir, labels_dir, prefix):

    if not images_dir.exists():
        print(f"WARNING: no imgs: {images_dir}")
        return

    if not labels_dir.exists():
        print(f"WARNING: no lbls: {labels_dir}")
        return

    for image_path in images_dir.iterdir():

        if image_path.suffix.lower() not in image_extensions:
            continue

        label_path = labels_dir / f"{image_path.stem}.txt"

        if not label_path.exists():
            print(f"WARNING: no label for {image_path.name}")
            continue

        lines = label_path.read_text().splitlines()

        fire_lines = []

        for line in lines:
            parts = line.split()

            if len(parts) >= 5 and parts[0] == "0":
                fire_lines.append(line)

        #  only take fire
        if not fire_lines:
            continue

        samples.append({
            "image": image_path,
            "label": label_path,
            "name": f"{prefix}_{image_path.stem}",
            "extension": image_path.suffix
        })

# ============================================================
# DATASET A
# train + valid + test
# ============================================================

for split in ["train", "val", "test"]:

    collect_dataset(
        dataset_A / split / "images",
        dataset_A / split / "labels",
        prefix=f"A_{split}"
    )

    if split=="val":
      split = "valid"

    collect_dataset(
        dataset_B / split / "images",
        dataset_B / split / "labels",
        prefix=f"B_{split}"
    )

print("\n================================")
print("Collected Samples")
print("================================")

print(f"Gesamt: {len(samples)}")

random.shuffle(samples)

if len(samples) < TRAIN_SIZE + VAL_SIZE:
    raise ValueError(
        f"only found: {len(samples)}, "
        f"need: {TRAIN_SIZE + VAL_SIZE}"
    )

selected_samples = samples[:TRAIN_SIZE + VAL_SIZE]

train_samples = selected_samples[:TRAIN_SIZE]
val_samples = selected_samples[TRAIN_SIZE:TRAIN_SIZE + VAL_SIZE]


def copy_samples(samples, split):

    for sample in samples:

        image_name = sample["name"] + sample["extension"]
        label_name = sample["name"] + ".txt"

        shutil.copy2(
            sample["image"],
            output / "images" / split / image_name
        )

        original_lines = sample["label"].read_text().splitlines()

        fire_lines = [
            line for line in original_lines
            if len(line.split()) >= 5 and line.split()[0] == "0"
        ]

        with open(output / "labels" / split / label_name, "w") as f:
            f.write("\n".join(fire_lines))


copy_samples(train_samples, "train")
copy_samples(val_samples, "val")


# ============================================================
# create data.yaml
# ============================================================

yaml_content = f"""path: {output}

train: images/train
val: images/val

nc: 1

names:
  0: Fire
"""

with open(output / "data.yaml", "w") as f:
    f.write(yaml_content)



print("\n================================")
print("MERGE COMPLETED")
print("================================")

print(f"Train: {len(train_samples)}")
print(f"Val:   {len(val_samples)}")
print(f"Total: {len(samples)}")

print("\nStructure:")

for path in sorted(output.rglob("*")):
    print(path)


Gesammelte Samples
Gesamt: 9201

MERGE ABGESCHLOSSEN
Train: 1000
Val:   200
Total: 9201

Struktur:
/content/fire_dataset/data.yaml
/content/fire_dataset/images
/content/fire_dataset/images/train
/content/fire_dataset/images/train/A_test_test_1020.jpg
/content/fire_dataset/images/train/A_test_test_1025.jpg
/content/fire_dataset/images/train/A_test_test_104.jpg
/content/fire_dataset/images/train/A_test_test_1049.jpg
/content/fire_dataset/images/train/A_test_test_1092.jpg
/content/fire_dataset/images/train/A_test_test_1103.jpg
/content/fire_dataset/images/train/A_test_test_1123.jpg
/content/fire_dataset/images/train/A_test_test_1127.jpg
/content/fire_dataset/images/train/A_test_test_1159.jpg
/content/fire_dataset/images/train/A_test_test_1160.jpg
/content/fire_dataset/images/train/A_test_test_1184.jpg
/content/fire_dataset/images/train/A_test_test_1191.jpg
/content/fire_dataset/images/train/A_test_test_1198.jpg
/content/fire_dataset/images/train/A_test_test_1217.jpg
/content/fire_dataset

In [ ]:
import shutil
from pathlib import Path

knife_dataset = Path("/content/knife_dataset")
fire_dataset = Path("/content/fire_dataset")

output = Path("/content/knife_fire_dataset")

if output.exists():
    shutil.rmtree(output)


for split in ["train", "val"]:
    (output / "images" / split).mkdir(parents=True, exist_ok=True)
    (output / "labels" / split).mkdir(parents=True, exist_ok=True)



def merge_dataset(dataset_path, prefix, class_mapping):

    for split in ["train", "val"]:

        images_dir = dataset_path / "images" / split
        labels_dir = dataset_path / "labels" / split

        if not images_dir.exists():
            print(f"WARNING: {images_dir} not found")
            continue

        if not labels_dir.exists():
            print(f"WARNING: {labels_dir} not found")
            continue

        for image_path in images_dir.iterdir():

            if image_path.suffix.lower() not in [
                ".jpg", ".jpeg", ".png",
                ".bmp", ".webp"
            ]:
                continue

            label_path = labels_dir / f"{image_path.stem}.txt"

            if not label_path.exists():
                continue

            new_image_name = f"{prefix}_{image_path.name}"
            new_label_name = f"{prefix}_{image_path.stem}.txt"

            shutil.copy2(
                image_path,
                output / "images" / split / new_image_name
            )

            new_labels = []

            for line in label_path.read_text().splitlines():

                parts = line.split()

                if len(parts) < 5:
                    continue

                old_class = int(parts[0])

                if old_class in class_mapping:

                    new_class = class_mapping[old_class]

                    parts[0] = str(new_class)

                    new_labels.append(" ".join(parts))

            with open(
                output / "labels" / split / new_label_name,
                "w"
            ) as f:

                f.write("\n".join(new_labels))

# ============================================================
# KNIFE DATASET
#
# 0 = Knife
# ============================================================

merge_dataset(
    knife_dataset,
    prefix="knife",
    class_mapping={
        0: 0
    }
)


# ============================================================
# FIRE DATASET
#
# 0 = Fire
#
# becomes:
#
# 1 = Fire
# ============================================================

merge_dataset(
    fire_dataset,
    prefix="fire",
    class_mapping={
        0: 1
    }
)


# ============================================================
# data.yaml
# ============================================================

yaml_content = f"""path: {output}

train: images/train
val: images/val

nc: 2

names:
  0: Knife
  1: Fire
"""

with open(output / "data.yaml", "w") as f:
    f.write(yaml_content)

print("\n================================")
print("MERGE COMPLETED")
print("================================")

for split in ["train", "val"]:

    images = list(
        (output / "images" / split).glob("*")
    )

    labels = list(
        (output / "labels" / split).glob("*.txt")
    )

    print(
        f"{split:5} | "
        f"Images: {len(images):5} | "
        f"Labels: {len(labels):5}"
    )

print("\nOutput:")
print(output)

print("\ndata.yaml:")
print((output / "data.yaml").read_text())


MERGE ABGESCHLOSSEN
train | Images:  1720 | Labels:  1720
val   | Images:   381 | Labels:   381

Output:
/content/knife_fire_dataset

data.yaml:
path: /content/knife_fire_dataset

train: images/train
val: images/val

nc: 2

names:
  0: Knife
  1: Fire



In [ ]:
from pathlib import Path
from collections import Counter

dataset = Path("/content/knife_fire_dataset")

for split in ["train", "val"]:

    counter = Counter()
    image_count = 0

    labels_dir = dataset / "labels" / split
    images_dir = dataset / "images" / split

    for label_file in labels_dir.glob("*.txt"):

        image_count += 1

        for line in label_file.read_text().splitlines():
            parts = line.split()

            if len(parts) >= 5:
                class_id = int(parts[0])
                counter[class_id] += 1

    print(f"\n========== {split.upper()} ==========")
    print(f"Knife (0): {counter[0]}")
    print(f"Fire  (1): {counter[1]}")
    print(f"Total Objects: {sum(counter.values())}")
    print(f"Images: {image_count}")


========== TRAIN ==========
Knife (0): 685
Fire  (1): 1077
Gesamt Objekte: 1762
Bilder: 1720

========== VAL ==========
Knife (0): 164
Fire  (1): 212
Gesamt Objekte: 376
Bilder: 381


##Using own data

In [ ]:
!unzip /content/cloudcomputing.v2i.yolov11.zip -d own_dataset

Archive:  /content/cloudcomputing.v2i.yolov11.zip
 extracting: own_dataset/train/images/WhatsApp-Video-2026-06-10-at-10_57_12_mp4-0018_jpg.rf.cda0aa96f83ddcd8aaa40dd1a3b6837f.jpg  
 extracting: own_dataset/train/images/WhatsApp-Video-2026-06-10-at-12_58_42-5-_mp4-0016_jpg.rf.9684a49cabdf90820eb7afd5c6e6d551.jpg  
 extracting: own_dataset/train/images/WhatsApp-Video-2026-06-10-at-12_58_42-2-_mp4-0026_jpg.rf.4af6363029a9ed4f420a8df76733097b.jpg  
 extracting: own_dataset/train/images/WhatsApp-Video-2026-06-10-at-10_57_12-3-_mp4-0007_jpg.rf.1da0a0625e3dd56e18f3fb51800c97b8.jpg  
 extracting: own_dataset/train/images/WhatsApp-Video-2026-06-10-at-11_05_00-2-_mp4-0001_jpg.rf.e5ee855215176893b9e4d2b19f54a4c3.jpg  
 extracting: own_dataset/train/images/WhatsApp-Video-2026-06-10-at-10_56_56_mp4-0025_jpg.rf.b2b627c71d5f203e507999c3796f0d12.jpg  
 extracting: own_dataset/train/images/WhatsApp-Video-2026-06-10-at-10_57_12-1-_mp4-0017_jpg.rf.5f273d886c565a07f5dee3ff26c77987.jpg  
 extracting: own_d

In [ ]:
keep = {
    0: 0, # fire
    1: 1, # knife
}

In [ ]:
from pathlib import Path
import shutil

src = Path("/content/own_dataset")
dst = Path("/content/own_filtered_dataset")

splits = ["train", "valid"]

for split in splits:
  img_src = src / split / "images"
  lbl_src = src / split / "labels"

  img_dst = dst / "images" / ("train" if split=="train" else "val")
  lbl_dst = dst / "labels" / ("train" if split=="train" else "val")

  img_dst.mkdir(parents=True, exist_ok=True)
  lbl_dst.mkdir(parents=True, exist_ok=True)

  for label_file in lbl_src.glob("*.txt"):
    new_lines  = []

    with open(label_file) as f:
      for line in f:
        parts = line.strip().split()
        cls = int(parts[0])

        if cls in keep:
          parts[0] = str(keep[cls])
          new_lines.append(" ".join(parts))

    if new_lines:
      out_label = lbl_dst / label_file.name
      with open(out_label, "w") as f:
        f.write("\n".join(new_lines))

      stem = label_file.stem
      for ext in [".jpg", ".jpeg", ".png"]:
        img = img_src / (stem + ext)
        if img.exists():
          shutil.copy2(img, img_dst / img.name)
          break

# Combine (same step for own data set, not shown here)


WARNING ⚠️ Box and segment counts should be equal, but got len(segments) = 98, len(boxes) = 416. To resolve this, only boxes will be used and all segments will be removed. To avoid this, please supply either a detect or segment dataset, not a detect-segment mixed dataset...Uncritical, since the training is for detection, not segmentation

In [ ]:
!unzip /content/knife.yolov11.zip -d knife_dataset

Die letzten 5000 Zeilen der Streamingausgabe wurden abgeschnitten.
  inflating: knife_dataset/train/labels/0000004920_jpg.rf.bByuIPkYBbvl8oMb6oIA.txt  
  inflating: knife_dataset/train/labels/0000001640_jpg.rf.67kNnmhw5tPv4dQhLdAn.txt  
  inflating: knife_dataset/train/labels/0000001413_jpg.rf.6AIqQAtIvvYElJ6ohHin.txt  
  inflating: knife_dataset/train/labels/1M3J5WA5HX1K_jpg.rf.GzCodAiM8QytDFom6eF4.txt  
  inflating: knife_dataset/train/labels/0000003704_jpg.rf.GpUQaaLZhGlw1lpDsYop.txt  
  inflating: knife_dataset/train/labels/0000000180_jpg.rf.H0j4zzLhFdfcFy9rQK7o.txt  
  inflating: knife_dataset/train/labels/0000001397_jpg.rf.H53Pbyu1HTeZp9Ik4u9z.txt  
  inflating: knife_dataset/train/labels/0000003764_jpg.rf.H0rWRUN4YlyrfHGf0JgA.txt  
  inflating: knife_dataset/train/labels/0000002233_jpg.rf.IpzIs0BsMHpvzuySAjpf.txt  
  inflating: knife_dataset/train/labels/0000003207_jpg.rf.J0UtIwiUiO4grTncczBp.txt  
  inflating: knife_dataset/train/labels/0000004797_jpg.rf.09P2P8RcpTdMUpn24kDA.tx

In [ ]:
!unzip knife_fire_dataset1.zip -d knife_fire_dataset

Archive:  knife_fire_dataset1.zip
   creating: knife_fire_dataset/content/knife_fire_dataset/
  inflating: knife_fire_dataset/content/knife_fire_dataset/data.yaml  
   creating: knife_fire_dataset/content/knife_fire_dataset/labels/
  inflating: knife_fire_dataset/content/knife_fire_dataset/labels/val.cache  
   creating: knife_fire_dataset/content/knife_fire_dataset/labels/train/
  inflating: knife_fire_dataset/content/knife_fire_dataset/labels/train/train_1714.txt  
  inflating: knife_fire_dataset/content/knife_fire_dataset/labels/train/032d284c9e52a9fa.txt  
  inflating: knife_fire_dataset/content/knife_fire_dataset/labels/train/WhatsApp-Video-2026-06-10-at-10_56_59_mp4-0011_jpg.rf.d3e1f8cfa5e08366fa31b72f9c9528f4.txt  
  inflating: knife_fire_dataset/content/knife_fire_dataset/labels/train/train_2166.txt  
  inflating: knife_fire_dataset/content/knife_fire_dataset/labels/train/train_1484.txt  
  inflating: knife_fire_dataset/content/knife_fire_dataset/labels/train/1b4bf12441f7e67c.t

In [ ]:
import os
import shutil
from pathlib import Path

SOURCE_DATASET = Path("/content/knife_dataset")
TARGET_DATASET = Path("/content/knife_fire_dataset/content/knife_fire_dataset")

# class in knife_dataset
SOURCE_KNIFE_CLASS = 3

# knife in our dataset
TARGET_KNIFE_CLASS = 0


def copy_knife_samples(source_split, target_split):
    source_images = SOURCE_DATASET / source_split / "images"
    source_labels = SOURCE_DATASET / source_split / "labels"

    target_images = TARGET_DATASET / "images" / target_split
    target_labels = TARGET_DATASET / "labels" / target_split

    target_images.mkdir(parents=True, exist_ok=True)
    target_labels.mkdir(parents=True, exist_ok=True)

    image_extensions = [".jpg", ".jpeg", ".png", ".bmp", ".webp"]

    copied = 0

    for image_path in source_images.iterdir():

        if image_path.suffix.lower() not in image_extensions:
            continue

        label_path = source_labels / f"{image_path.stem}.txt"

        if not label_path.exists():
            continue

        knife_annotations = []

        with open(label_path, "r") as f:
            for line in f:
                parts = line.strip().split()

                if len(parts) < 5:
                    continue

                class_id = int(parts[0])

                # only keep knife
                if class_id == SOURCE_KNIFE_CLASS:
                    # class 3 -> class 0
                    parts[0] = str(TARGET_KNIFE_CLASS)
                    knife_annotations.append(" ".join(parts))

        if not knife_annotations:
            continue

        #prefix e
        new_stem = f"e_{source_split}_{image_path.stem}"

        target_image = target_images / f"{new_stem}{image_path.suffix}"
        target_label = target_labels / f"{new_stem}.txt"

        shutil.copy2(image_path, target_image)


        with open(target_label, "w") as f:
            f.write("\n".join(knife_annotations) + "\n")

        copied += 1

    print(
        f"{source_split} -> {target_split}: "
        f"{copied} Knife-Samples kopiert"
    )

copy_knife_samples("train", "train")


print("Finished.")

train -> train: 5769 Knife-Samples kopiert
Fertig.


In [ ]:
from collections import Counter
from pathlib import Path

TARGET_DATASET = Path("/content/knife_fire_dataset/content/knife_fire_dataset")

CLASS_NAMES = {
    0: "Knife",
    1: "Fire"
}


def analyze_split(split):
    labels_dir = TARGET_DATASET / "labels" / split
    images_dir = TARGET_DATASET / "images" / split

    class_counter = Counter()
    image_count = 0
    annotation_count = 0

    image_extensions = {".jpg", ".jpeg", ".png", ".bmp", ".webp"}

    for image_path in images_dir.iterdir():
        if image_path.suffix.lower() in image_extensions:
            image_count += 1

    for label_path in labels_dir.glob("*.txt"):

        with open(label_path, "r") as f:
            for line in f:
                parts = line.strip().split()

                if len(parts) < 5:
                    continue

                class_id = int(parts[0])

                class_counter[class_id] += 1
                annotation_count += 1

    print(f"\n{'=' * 40}")
    print(f"{split.upper()}")
    print(f"{'=' * 40}")

    print(f"Images:       {image_count}")
    print(f"Annotations: {annotation_count}")

    for class_id, class_name in CLASS_NAMES.items():
        count = class_counter[class_id]

        percentage = (
            count / annotation_count * 100
            if annotation_count > 0
            else 0
        )

        print(
            f"{class_name:10s}: "
            f"{count:6d} "
            f"({percentage:6.2f} %)"
        )

analyze_split("train")
analyze_split("val")


TRAIN
Bilder:       4288
Annotationen: 4899
Knife     :   4000 ( 81.65 %)
Fire      :    899 ( 18.35 %)

VAL
Bilder:       194
Annotationen: 224
Knife     :    121 ( 54.02 %)
Fire      :    103 ( 45.98 %)


with about 1000 Fire images we could already get good results, the knife needed more data → to cover the training inbalance, the confidence threshold is adapted for each class

In [ ]:
!zip -r knife_fire_dataset.zip /content/knife_fire_dataset/content/knife_fire_dataset,

Die letzten 5000 Zeilen der Streamingausgabe wurden abgeschnitten.
  adding: content/knife_fire_dataset/content/knife_fire_dataset/labels/train/010dce7cb9251d5e.txt (deflated 11%)
  adding: content/knife_fire_dataset/content/knife_fire_dataset/labels/train/e_train_0000002684_jpg.rf.tMwaEQijEerFqnwI2kNw.txt (deflated 42%)
  adding: content/knife_fire_dataset/content/knife_fire_dataset/labels/train/e_train_0000003971_jpg.rf.JQpHeO2CDA9IndJS1qJ2.txt (deflated 11%)
  adding: content/knife_fire_dataset/content/knife_fire_dataset/labels/train/e_train_0000003731_jpg.rf.08eziB5h8EIonkCf0SLm.txt (deflated 15%)
  adding: content/knife_fire_dataset/content/knife_fire_dataset/labels/train/26281ba127e953b1.txt (deflated 43%)
  adding: content/knife_fire_dataset/content/knife_fire_dataset/labels/train/e_train_0000001243_jpg.rf.nMkKNehPDPKiPaRcxsfx.txt (deflated 35%)
  adding: content/knife_fire_dataset/content/knife_fire_dataset/labels/train/e_train_4I172SJ9GDYH_jpg.rf.R8e3Sm1jG7TsHyJw1YGO.txt (defl

In [ ]:
from ultralytics import YOLO

model = YOLO("yolo11n.pt")

model.train(
    data="/content/knife_fire_dataset/content/knife_fire_dataset/data.yaml",

    epochs=100,
    imgsz=640,

    # ========================================================
    # Lighting / Color
    # ========================================================

    hsv_h=0.015,
    hsv_s=0.45,
    hsv_v=0.25,

    # ========================================================
    # Geometric Augmentation
    # ========================================================

    # Especially important for knives at different angles
    degrees=25,

    # Vary the object's position in the image
    translate=0.10,

    # different object sizes
    scale=0.35,

    # slight change in perspective
    shear=3.0,
    perspective=0.0005,

    # ========================================================
    # Reflection
    # ========================================================

    fliplr=0.5,
    flipud=0.0,

    # ========================================================
    # Multi-object / Different Image Compositions
    # ========================================================

    mosaic=0.5,
    mixup=0.0,

    # ========================================================
    # Training
    # ========================================================

    batch=-1,
    device=-1,
    patience=10,
    verbose=True
)

Searching for 1 idle GPUs with free memory >= 20.0% and free utilization >= 0.0%...
Selected idle CUDA devices [0]
Ultralytics 8.4.126 🚀 Python-3.13.15 torch-2.11.0+cu128 CUDA:0 (NVIDIA L4, 22563MiB)
engine/trainer: agnostic_nms=False, amp=True, angle=1.0, augment=False, auto_augment=randaugment, batch=-1, bgr=0.0, box=7.5, cache=False, cfg=None, channels_last=False, classes=None, close_mosaic=10, cls=0.5, cls_pw=0.0, cls_remap=True, compile=False, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=/content/knife_fire_dataset/content/knife_fire_dataset/data.yaml, degrees=25, deterministic=True, device=0, dfl=1.5, dgrad=0.5, dis=6.0, distill_model=None, dlam=1.0, dlog=1.0, dnn=False, dropout=0.0, dynamic=False, embed=None, end2end=None, epochs=100, erasing=0.4, exist_ok=False, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, hsv_h=0.015, hsv_s=0.45, hsv_v=0.25, imgsz=640, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=0.01, lrf=0.

ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7ece8c19db70>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.04804

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95):
                   all        194        224      0.947      0.878       0.96      0.672
                 Knife        100        121      0.956      0.884      0.964      0.746
                  Fire         94        103      0.937      0.873      0.955      0.599
    shows that although fire had way less training data it nearly equals the detection metrics mAP50 of Knife. But the Intersection over Union (mAP50-95) is worse, meaning the quality of bounding boxes could still be improved.

In [ ]:
model_human = YOLO("best_human.pt")
model_human.export(format="onnx", opset=11)

Ultralytics 8.4.126 🚀 Python-3.13.15 torch-2.11.0+cu128 CPU (Intel Xeon CPU @ 2.20GHz)
Model summary (fused): 72 layers, 3,005,843 parameters, 0 gradients, 8.1 GFLOPs

PyTorch: starting from 'best_human.pt' with input shape (1, 3, 640, 640) BCHW and output shape(s) (1, 5, 8400) (6.0 MB)

ONNX: starting export with onnx 1.22.0 opset 11...
ONNX: slimming with onnxslim 0.1.96...
ONNX: export success ✅ 1.0s, saved as 'best_human.onnx' (11.7 MB)

Export complete (1.2s)
Results saved to /content/best_human.onnx
Predict:         yolo predict task=detect model=best_human.onnx imgsz=640 
Validate:        yolo val task=detect model=best_human.onnx imgsz=640 data=/kaggle/input/v4-yaml/data.yaml  
Visualize:       https://netron.app


'best_human.onnx'